configure.ac

```diff
@@ -402,20 +391,16 @@ if test "${shapelib_desired}" = "yes" -a "${force_internal_shapelib}" = "no"; th
 fi
 if test "${shapelib_desired}" = "yes" -a "${use_shapelib}" = "no"; then
  AC_MSG_WARN([**************************************************************** ])  
- AC_MSG_WARN([Your system does not have shapelib installed.  Using an internal ])
- AC_MSG_WARN([version.  This may lead to a larger filesize for the executable. ])
+ AC_MSG_WARN([Your system does not have shapelib installed.                    ])
  AC_MSG_WARN([Install shapelib on your system to eliminate this warning.       ])
  AC_MSG_WARN([**************************************************************** ])  
  LIBSHP="-Lshapelib -lshape"
  CPPFLAGS="-I\$(top_srcdir)/src/shapelib $CPPFLAGS"
- SUBSHP="shapelib" 
- use_shapelib=yes
- force_internal_shapelib=yes
+ use_shapelib=no
  AC_DEFINE(HAVE_DBFGETFIELDINDEX, , Define to 1 if your `shp' library has DBFGetFieldIndex. ) 
  AC_DEFINE(HAVE_LIBSHP, , Define to 1 if you have the `shp' library (-lshp). )
  AC_DEFINE(HAVE_SHAPEFIL_H, , Define to 1 if you have the `shapefil.h' header. )
```

Comment:
I believe this is in error.

It retains the setting of HAVE_LIBSHP, HAVE_SHAPEFIL_H, and HAVE_DBFGETFIELDINDEX even while saying we don't have shapelib at all.

Reply:
That is, if we're going to allow configure to proceed with a notice to install shapelib to silence a notification, we should also make the build work.  And it won't as it is.

I think if you remove the three AC_DEFINEs, it'll be right.

Reply:
Actually, I think you have to remove everything other than the warning here --- including the setting of LIBSHP and CPPFLAGS.   Only the warning text and the setting of use_shapelib=no should be inside the conditional there.  Then it should once again be possible to build Xastir without shapelib support (and have it really crippled).

If you want it to be a *mandatory* TPL, then configure should emit an *ERROR* about shapelib being unavailable, and not leave it to compile time to fail.

Reply:
More comments.  Sorry. 

If you look at the LIBSHP variable, its entire purpose is to handle passing the internal shapelib -l/-L flags around.  So it should be removed from configure.ac (along with the AC_SUBST commands that pass it down to Makefile.ams), and should also be removed from src/Makefile.am in two places (where it appears as "@LIBSHP@"